### DISCUSSION TITLE
how to use *_init functions

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [libuv/libuv#3665] [Q&A] [answered]
How to use *_init functions in libuv? Should i initialize memory using malloc for specific handle and then pass it to *_init functions or *_init functions do this for us? For example I saw an example that you don't need to use malloc for initializing uv_loop_t:
```
uv_loop_t loop;
uv_loop_init(&loop);
```

**Comment by bnoordhuis** [comment, marked as answer]
You're responsible for providing storage (i.e., memory.)

You can use dynamic (malloc), static, or on-stack storage, as long as the memory stays around until uv_close() calls your close callback. Only then is it safe to deallocate it.

**Reply by MadadiSaeed** [reply]
Thanks for the response.
